src/main/java/io/reactivesocket/cli/Main.java

```diff
@@ -241,6 +317,26 @@ private Completable runSingleOperation(ReactiveSocket client) {
         return ObservableIO.lines(is);
     }
 
+    private String getMetadata() {
+        synchronized (this) {
+            if (cachedMetadata == null) {
+                if (metadata == null) {
+                    cachedMetadata = "";
+                } else if (metadata.startsWith("@")) {
```

Comment:
Looks like this block can be created as a utility function (to read data from a file)